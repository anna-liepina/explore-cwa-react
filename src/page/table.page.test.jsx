import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import TablePage from './table.page';

describe('<TablePage/>', () => {
    const props = {
        form: { config: [] },

        columns: [
            {
                label: 'date',
                key: 'date',
            },
            {
                label: 'price',
                key: 'price',
            },
            {
                label: 'address',
                key: 'address',
            },
        ]
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        perPage: 250,
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<TablePage {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<TablePage {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
