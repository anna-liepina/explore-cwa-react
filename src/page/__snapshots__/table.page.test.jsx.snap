// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePage/> render with default/required props 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form_controls"
    />
  </form>
  <div
    class="pagination"
  >
    <button
      class="button"
      data-cy="-top-pagination-prevpage"
      disabled=""
    >
      Previous
    </button>
    <span>
      Page 1 of unknown
    </span>
    <button
      class="button"
      data-cy="-top-pagination-nextpage"
    >
      Next
    </button>
  </div>
  <div
    class="table-container"
  >
    <div
      class="table-stats"
    />
    <div
      class="table-header"
    >
      <div
        class="table-cell"
      >
        date
      </div>
      <div
        class="table-cell"
      >
        price
      </div>
      <div
        class="table-cell"
      >
        address
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TablePage/> render with optional/required props 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form_controls"
    />
  </form>
  <div
    class="pagination"
  >
    <button
      class="button"
      data-cy="optProps.data-cy-top-pagination-prevpage"
      disabled=""
    >
      Previous
    </button>
    <span>
      Page 1 of unknown
    </span>
    <button
      class="button"
      data-cy="optProps.data-cy-top-pagination-nextpage"
    >
      Next
    </button>
  </div>
  <div
    class="table-container optProps.className"
  >
    <div
      class="table-stats"
    />
    <div
      class="table-header"
    >
      <div
        class="table-cell"
      >
        date
      </div>
      <div
        class="table-cell"
      >
        price
      </div>
      <div
        class="table-cell"
      >
        address
      </div>
    </div>
  </div>
</DocumentFragment>
`;
